tutorials/future/jax/datasets.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Datasets used in examples."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import array
+import gzip
+import os
+from os import path
+import struct
+from six.moves.urllib.request import urlretrieve
+
+import numpy as np
+
+
+_DATA = "/tmp/jax_example_data/"
+
+
+def _download(url, filename):
+  """Download a url to a file in the JAX data temp directory."""
+  if not path.exists(_DATA):
+    os.makedirs(_DATA)
+  out_file = path.join(_DATA, filename)
+  if not path.isfile(out_file):
+    urlretrieve(url, out_file)
+    print("downloaded {} to {}".format(url, _DATA))
+
+
+def _partial_flatten(x):
+  """Flatten all but the first dimension of an ndarray."""
+  return np.reshape(x, (x.shape[0], -1))
+
+
+def _one_hot(x, k, dtype=np.float32):
```

Comment:
May be reuse one_hot from `jax/utils` and also it could make sense to move `partial_flatten` there too.